src/Action/RequestAction.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\UserBundle\Action;
+
+use Sonata\AdminBundle\Admin\Pool;
+use Sonata\AdminBundle\Templating\TemplateRegistryInterface;
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+
+final class RequestAction
+{
+    /**
+     * @var EngineInterface
+     */
+    private $templating;
+
+    /**
+     * @var UrlGeneratorInterface
+     */
+    private $urlGenerator;
+
+    /**
+     * @var AuthorizationCheckerInterface
+     */
+    private $authorizationChecker;
+
+    /**
+     * @var Pool
+     */
+    private $adminPool;
+
+    /**
+     * @var TemplateRegistryInterface
+     */
+    private $templateRegistry;
+
+    public function __construct(
+        EngineInterface $templating,
+        UrlGeneratorInterface $urlGenerator,
+        AuthorizationCheckerInterface $authorizationChecker,
+        Pool $adminPool,
+        TemplateRegistryInterface $templateRegistry
+    ) {
+        $this->templating = $templating;
+        $this->urlGenerator = $urlGenerator;
+        $this->authorizationChecker = $authorizationChecker;
+        $this->adminPool = $adminPool;
+        $this->templateRegistry = $templateRegistry;
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        if ($this->authorizationChecker->isGranted('IS_AUTHENTICATED_FULLY')) {
+            return new RedirectResponse($this->urlGenerator->generate('sonata_admin_dashboard'));
+        }
+
+        return $this->templating->renderResponse('@SonataUser/Admin/Security/Resetting/request.html.twig', [
+            'base_template' => $this->templateRegistry->getTemplate('layout'),
+            'admin_pool' => $this->adminPool,
```

Comment:
Not part of this PR, but can we make a global twig variable for `base_template` and `admin_pool`, as every admin template need this